### ISSUE TITLE
[bug]`Fast playout to the end ` exception

### ISSUE THREAD

**Issue by xiaoyifang** [sanderland/katrain#675]
Please read and fill in the appropriate template

# I have a bug report

Steps: 
1, press `Fast playout to the end`
2, while the node is still generating ,right click the new added node and choose ,expand .
3, error throw.

![image](https://github.com/sanderland/katrain/assets/105986/b89ffaa9-6be7-4ed7-a399-187259985b6c)

```

ERROR: Exception KeyError: SGFNode({'W': ['bm']})
  File "kivy\_clock.pyx", line 649, in kivy._clock.CyClockBase._process_events
  File "kivy\_clock.pyx", line 218, in kivy._clock.ClockEvent.tick
  File "G:\katrain\katrain\gui\widgets\movetree.py", line 228, in <lambda>
    lambda _dt: self.move_tree_canvas.draw_move_tree(self.current_node, self.insert_node), 0.1
  File "G:\katrain\katrain\gui\widgets\movetree.py", line 157, in draw_move_tree
    x = self.move_pos[parent][0] + 1
```

open the generated sgf file also can trigger this exception.

[KaTrain_韩一洲 五段 vs Master play to the end.zip](https://github.com/sanderland/katrain/files/13810187/KaTrain_.vs.Master.play.to.the.end.zip)

I guess the action I took has damaged the sgf file in some way.

This can be reproduced with other sgf files.

**Comment by sanderland** [comment]
Confirmed, though I haven't figured out why yet.

**Comment by HackYardo** [comment]
I have no idea to fix it, but the SGF is a "plain text format", it can be read and wrote by any text exitor such as Notepad, Nano, Sublime, VSCode after added a "txt" suffix, e.g. "**xxx.sgf.txt**". So there're no "damaged sgf file" usually.

**Comment by xiaoyifang** [comment]
I guess there is a concurrent issue, when the  `fast playout` generating the nodes , the manual operation has conflict with the previous operation.

**Comment by sanderland** [comment]
Hopefully resolved with [32b3ad9](https://github.com/sanderland/katrain/pull/648/commits/32b3ad9f4b794d705da6697fbf7a965a3fb2363e)
Still not 100% sure what is causing it, probably the rapid removal/adding of shortcuts in fast playouts

**Comment by xiaoyifang** [comment]
I noticed that:
If  modify the sgf manually and removed all the KTSID node
![image](https://github.com/sanderland/katrain/assets/105986/5143effd-181e-4821-8e27-7317b7efba0a)

,it can also work normally.
